### PR TITLE
fix #191 Add Users view

### LINKED ISSUE #191: Add support for Users management
ACL should be added to manage users

### PR DESCRIPTION
**Pull request by gkorland**
## Summary by CodeRabbit

- **New Features**
  - Introduced new user management API routes, including retrieval, creation, and deletion of users.
  - Added new React components for adding and deleting users, and managing user lists with selection capabilities.
  - Enhanced navigation by adding a "Users" section in the navbar.

- **Refactor**
  - Updated authentication logic across various API routes, centralizing client connection and authentication checks.
  
- **Bug Fixes**
  - Improved error handling in API routes to ensure robust client response checks.

- **Documentation**
  - None explicitly mentioned, but changes are implied in the use of new components and API routes.

- **Style**
  - Added a new `Checkbox` component with custom styling and behavior.

- **Chores**
  - Updated package dependencies to include `@radix-ui/react-checkbox`. 

- **Tests**
  - None mentioned.

- **Revert**
  - None mentioned.

### CODE REVIEW

**Comment by gkorland** [comment]
@coderabbitai review

File: app/api/auth/[...nextauth]/options.ts [hunk 1]
```diff
@@ -121,6 +109,30 @@ const authOptions: AuthOptions = {
...
+            host: user.host,
+            port: user.port.toString() ?? "6379",
+            username: user.username,
+            password: user.password,
+        }, user.id)
+    }
+
+    if(!client) {
+        return NextResponse.json({ message: "Not authenticated" }, { status: 401 })
+    }
```
[the remark this answers is not in the document]
Comment (gkorland) [reply app/api/auth/[...nextauth]/options.ts:134]: It's not the same functionality,

**Comment by gkorland** [comment]
@coderabbitai review

**Review by Anchel123 — DISMISSED** [review]

**Review by Anchel123 — COMMENTED** [review]
the delete user throw a type error

**Review by Anchel123 — APPROVED** [review]

**Pull request merged**